src/modules/trajectory_generation/domain/spline_generation/cubic_spline_generator.hpp

```diff
@@ -2,19 +2,160 @@
 #define CUBIC_SPLINE_GENERATOR_HPP
 
 #include <i_spline_generator.hpp>
-#include <Posture.hpp>
+#include <eigen3/Eigen/Dense>
 
 namespace trajectory_generation {
-    class CubicSplineGenerator : public ISplineGenerator{
+    /**
+     * @brief CubicSplineGenerator Class for generating cubic splines given two postures
+     * 
+     */
+    class CubicSplineGenerator {
         public:
-            CubicSplineGenerator(const core_datastructures::Posture& start, const core_datastructures::Posture& goal);
-            /*
-                Generate splines as per given end posture
-                @param[in] core_datastructures::Posture posture
-            */
-            std::vector<core_datastructures::Posture> generate_splines(core_datastructures::Posture posture);
+            /**
+             * @brief Construct a new Cubic Spline Generator object
+             * 
+             * @param start core_datastructures::Posture spline start posture
+             * @param goal core_datastructures::Posture spline end posture
+             */
+            CubicSplineGenerator(core_datastructures::Posture& start, core_datastructures::Posture& goal);
+            
+            /**
+             * @brief Get the spline object
+             * 
+             * @return std::vector<core_datastructures::Posture> containing spline points
+             */
+            std::vector<core_datastructures::Posture> get_spline();
+
+            /**
+             * @brief Computes the spline parameters given [P1, P2, SG] values
+             * 
+             * @param p1 
+             * @param p2 
+             * @param s 
+             * @return std::vector<double> the spline parameters {a, b, c, d}
+             */
+            std::vector<double> calculate_parameters(double p1, double p2, double s);
+
+            /**
+             * @brief Get the next state object
+             * 
+             * @param params 
+             * @param s 
+             * @return core_datastructures::Posture return next state given spline params
+             */
+            core_datastructures::Posture get_next_state(std::vector<double>& params, double s);
+
+
+            /**
+             * @brief Computes Jacobian matrix for the given params
+             * 
+             * @param params 
+             * @return Eigen::Matrix3d Jacobian Matrix
+             */
+            Eigen::Matrix3d calculate_Jacobian(std::vector<double>& params);
```

Comment:
Maybe spline_coeffs is a good variable name. Just a suggestion. Feel free to resolve.